framework/base/src/api/managed_types/static_var_api.rs

```diff
@@ -31,4 +31,12 @@ pub trait StaticVarApiImpl {
     fn set_call_value_multi_esdt_handle(&self, handle: RawHandle);
 
     fn get_call_value_multi_esdt_handle(&self) -> RawHandle;
+
+    fn set_scaling_factor_init(&self, scaling_factor: [bool; const_handles::SCALING_FACTOR_LENGTH as usize]);
+
+    fn get_scaling_factor_init(&self) -> [bool; const_handles::SCALING_FACTOR_LENGTH as usize];
```

Comment:
It's a lot simpler than this, you don't need to return the whole thing. The data can remain hidden.
`fn is_scaling_factor_cached(&self, decimals: usize) -> bool` is enough